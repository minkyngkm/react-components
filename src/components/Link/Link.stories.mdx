import { ArgsTable, Canvas, Meta, Story } from "@storybook/addon-docs/blocks";

import Link from "./Link";
import Strip from "../Strip";

<Meta
  title="Link"
  component={Link}
  argTypes={{
    children: {
      control: {
        type: "text",
      },
    },
  }}
/>

export const Template = (args) => <Link {...args} />;

### Link

This is a [React](https://reactjs.org/) component for the Vanilla [Link](https://docs.vanillaframework.io/patterns/links/).

Links are used to embed actions or pathways to more information, allowing users to click their way from page to page.

### Props

<ArgsTable of={Link} />

### Default

<Canvas>
  <Story
    name="Default"
    args={{
      children:
        "Fast, secure and simple, Ubuntu powers millions of PCs worldwide",
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

### Soft

<Canvas>
  <Story name="Soft">
    <Link soft>Learn about MAAS</Link>
  </Story>
</Canvas>

### Inverted

<Canvas>
  <Story name="Inverted">
    <Strip type="dark">
      <Link inverted>Package &amp; publish your app</Link>
    </Strip>
  </Story>
</Canvas>

### Back to top

<Canvas>
  <Story name="Back to top">
    <Link top>Back to top</Link>
  </Story>
</Canvas>
