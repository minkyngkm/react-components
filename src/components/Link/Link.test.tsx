import { shallow } from "enzyme";
import React from "react";

import Link from "./Link";

describe("Link ", () => {
  it("renders", () => {
    const wrapper = shallow(<Link href="/example">Test content</Link>);
    expect(wrapper).toMatchSnapshot();
  });

  it("can be an inverted link", () => {
    const wrapper = shallow(<Link inverted={true}>Test content</Link>);
    expect(wrapper.prop("className").includes("p-link--inverted")).toBe(true);
  });

  it("can be a soft link", () => {
    const wrapper = shallow(<Link soft={true}>Test content</Link>);
    expect(wrapper.prop("className").includes("p-link--soft")).toBe(true);
  });

  it("can be a back to top link", () => {
    const wrapper = shallow(<Link top={true}>Back to top</Link>);
    expect(wrapper).toMatchSnapshot();
  });

  it("can add additional classes", () => {
    const wrapper = shallow(
      <Link soft={true} className="extra-class">
        Test content
      </Link>
    );
    const className = wrapper.prop("className");
    expect(className.includes("p-link--soft")).toBe(true);
    expect(className.includes("extra-class")).toBe(true);
  });
});
