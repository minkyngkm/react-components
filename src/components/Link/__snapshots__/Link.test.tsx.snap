// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link  can be a back to top link 1`] = `
<div
  className="p-top"
>
  <a
    className="p-top__link"
    href="#"
  >
    Back to top
  </a>
</div>
`;

exports[`Link  renders 1`] = `
<a
  className=""
  href="/example"
>
  Test content
</a>
`;
